projects/kit/components/input-slider/input-slider.component.ts

```diff
@@ -0,0 +1,145 @@
+        }),
+    ],
+    host: {
+        '(input)': 'onSliderInput($event)',
+        '(change)': 'onSliderInput($event)',
+        '(click)': 'textfield.input.nativeElement.focus()',
+        '(focusout)': 'onBlur()',
+        '(keydown.arrowUp)': 'onStep(1)',
+        '(keydown.arrowDown)': 'onStep(-1)',
+    },
+})
+export class TuiInputSliderDirective {
+    private readonly injector = inject(INJECTOR);
+    private keyStepsTransformer: WritableSignal<TuiValueTransformer<
+        number,
+        number
+    > | null> = signal(null);
+
+    private readonly inputNumber = signal<TuiInputNumber | null>(null);
+    private readonly slider = signal<TuiSliderComponent | null>(null);
+    protected readonly textfield = inject(TuiTextfieldComponent);
+
+    protected min = computed(() => this.inputNumber()?.min() ?? 0);
+    protected max = computed(() => this.inputNumber()?.max() ?? 100);
+    protected step = computed((slider = this.slider()) =>
+        slider && this.keyStepsTransformer() // For non-linear slider step means percentage
+            ? ((this.max() - this.min()) / 100) * slider.step()
+            : (slider?.step() ?? 1),
+    );
+
+    protected disabled = computed(
+        () => (this.inputNumber()?.disabled() || this.inputNumber()?.readOnly()) ?? false,
+    );
+
+    protected safeInputNumberValue = toSignal(
+        toObservable(this.inputNumber).pipe(
+            filter(tuiIsPresent),
+            switchMap(({value}) =>
+                toObservable(value, {injector: this.injector}).pipe(startWith(value())),
+            ),
+            pairwise(),
+            map(([prev, cur]) => cur ?? prev!),
+        ),
+    );
+
+    protected readonly textfieldToSliderSync = effect(() => {
+        const slider = this.slider();
+
+        if (!slider) {
+            return;
+        }
+
+        const value = this.safeInputNumberValue() ?? slider.value;
+
+        slider.value = this.keyStepsTransformer()?.fromControlValue(value) ?? value;
+        slider.el.disabled = this.disabled();
+    }, TUI_ALLOW_SIGNAL_WRITES);
+
+    @ContentChild(TuiInputNumber)
+    protected set inputNumberSetter(x: TuiInputNumber) {
+        this.inputNumber.set(x);
+    }
+
+    @ContentChild(TuiSliderComponent)
+    protected set sliderSetter(slider: TuiSliderComponent) {
+        this.slider.set(slider);
+        slider.min = this.min;
+        slider.max = this.max;
+        slider.computedStep = this.step;
+        slider.el.setAttribute('tabindex', '-1');
+    }
+
+    @ContentChild(TuiSliderKeyStepsBase)
+    protected set keyStepsSetter({transformer}: TuiSliderKeyStepsBase) {
+        this.keyStepsTransformer = transformer;
+    }
+
+    protected onSliderInput({target}: Event): void {
+        if (tuiIsElement(target) && tuiIsInput(target) && target.type === 'range') {
+            const value = Number(target.value);
+
+            this.inputNumber()?.setValue(
+                this.keyStepsTransformer()?.toControlValue(value) ?? value,
```

Comment:
nit: How about we write such things like:
```ts
const transformed = this.keyStepsTransformer()?.toControlValue(value) ?? value;

this.inputNumber()?.setValue(transformed);
```
That's 1 line shorter and more readable.